src/sql/workbench/api/node/extHostModelView.ts

```diff
@@ -777,10 +778,10 @@ class DropDownWrapper extends ComponentWrapper implements sqlops.DropDownCompone
 		this.setProperty('value', v);
 	}
 
-	public get values(): string[] {
+	public get values(): string[] | sqlops.CategoryValue[] {
 		return this.properties['values'];
 	}
-	public set values(v: string[]) {
+	public set values(v: string[] | sqlops.CategoryValue[]) {
```

Comment:
As I understood the request in #1621, instead of category value it should be something more like the following interface:
`public set values<T>(v: string[] | { value: T, displayName: string })`

That way the user can make any arbitrary object correspond with the display name, not just strings. This could be a bit tricky to implement though since we can't pass arbitrary objects from the extension host to the main thread, but we should just be able to keep track of them in the extension host.

Reply:
I don't think I want to support any type, they should be only string [] or with display name

---
In reply to: [195883920](https://github.com/Microsoft/sqlopsstudio/pull/1640#discussion_r195883920) [](ancestors = 195883920)

Reply:
@kevcunnane can you clarify what you need this for? To me accepting arbitrary types makes more sense than just accepting strings, but extension authors could always just keep a map on their side so it's probably not that important